guides/integrate-nextauth.md

```diff
@@ -0,0 +1,327 @@
+
+<AppSchemeCaution/>
+
+3. Set the the [Invocation Type](/docs/v1/platform-overview/authenticator-config#invocation-type). This specifies how our authentication URL is delivered to your application. Invocation Type can be one of two values:
+
+- **Automatic**: redirect to your application using the Invoke URL with a challenge that your app will need to sign.
+
+- **Manual**: the challenge will be returned to you as part of a JSON response.
+
+<InvocationTip/>
+
+For **NextAuth**, we will use **Automatic**.
+
+<InvocationDiagram />
+
+![Invocation Type](../docs/workflows/screenshots/authentication-invocation.png)
+
+### Start Authorization for Invocation Type
+
+The authenticate url that is redirected to your application will append a `/bi-authenticate` path to your Invoke URL. Use a "/bi-authenticate" route to intercept this url in your application:
+
+```
+$invoke_url/bi-authenticate?request=<request>
+```
+
+#### Handle Authorization in your applicaiton
+
+The authorization flow should begin in a webview with your [crafted authorization URL](/docs/v1/workflows/authentication#1-craft-authorization-url).
+
+Don't forget to [initalize your SDK](/docs/v1/workflows/sdk-setup) and present some logic to the user to select a passkey ahead of time.
+
+:::info
+In a native application, the webview used should follows best practices set out in [RFC 8252 - OAuth 2.0 for Native Apps](https://www.rfc-editor.org/rfc/rfc8252). This means that on iOS applications [ASWebAuthenticationSession](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession) should be used, and [Custom Tabs](https://developer.chrome.com/docs/android/custom-tabs/) should be used on Android devices.
+
+`UIWebView`, `WKWebView`, and `WebView` are explicitly not supported due to the security and usability reasons explained in [Section 8.12 of RFC 8252](https://www.rfc-editor.org/rfc/rfc8252#section-8.12).
+:::
+
+1. Configuring NextAuth
+
+Under `next-auth-example/pages/api/auth/[...nextauth].ts`, add the following Beyond Identity provider. The provider will go through an OAuth/OIDC that will result in fetching an id token that will log you in to the example app. Many of the values used to configure the request are the values from your [crafted authorization URL](/docs/v1/workflows/authentication#1-craft-authorization-url).
+
+```javascript
+...
+import BeyondIdentity from "@auth/core/providers/beyondidentity"
+...
+providers: [
+  //@ts-expect-error issue https://github.com/nextauthjs/next-auth/issues/6174
+  BeyondIdentity({
+    clientId: process.env.BEYOND_IDENTITY_CLIENT_ID,
+    clientSecret: process.env.BEYOND_IDENTITY_CLIENT_SECRET,
+    issuer: process.env.BEYOND_IDENTITY_ISSUER,
```

Comment:
What would someone put here for the issuer? I don't think I had set that in Acme so I'm not actually sure what that would be.

Reply:
Yes, there is an Issuer in the Console, in the App. It looks like https://auth-{REGION}.beyondidentity.com/v1/tenants/{TENANT_ID}/realms/{REALM_ID}/applications/{APPLICATION_ID}.

Acme is currently setting:
```
providers: [BeyondIdentity({ clientId: clientId, clientSecret: clientSecret, issuer: `${extendedAuthUrl(
  tenant.region,
  tenant.id,
  tenant.realmId,
  tenant.webApplicationId
).toString()}` })],
```

Reply:
I do call it out in one spot. At the bottom of "Create an Application". Should I add it here, along with the Client ID and Secret instructions?
```
1. For `BeyondIdentity` _issuer_, copy and paste the value from Applications -> Your New Application -> External Protocol -> Issuer
```

Reply:
Ah got it, I see. I think we can remind them that the values were added to the environment variables above. I added two new comments for that.